File: android/src/main/java/com/lib/flutter_blue_plus/l2cap/messages/WriteL2CapChannelRequest.java
// Copyright 2024, Continental Automotive Technologies GmbH
// All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.
package com.lib.flutter_blue_plus.l2cap.messages;

import com.lib.flutter_blue_plus.MarshallingUtil;
import com.lib.flutter_blue_plus.l2cap.L2CapAttributeNames;

import java.util.Map;

public class WriteL2CapChannelRequest {
    public final String remoteId;
    public final int psm;
    public final byte[] value;

    public WriteL2CapChannelRequest(String remoteId, int psm, byte[] value) {
        this.remoteId = remoteId;
        this.psm = psm;
        this.value = value;
    }

    public static WriteL2CapChannelRequest unmarshal(final Map<String, Object> data) {
        final int psm = (int) data.get(L2CapAttributeNames.KEY_PSM);
        final String remoteId = (String) data.get(L2CapAttributeNames.KEY_REMOTE_ID);
        final String valueAsString = (String) data.get(L2CapAttributeNames.KEY_VALUE);
        return new WriteL2CapChannelRequest(remoteId, psm, MarshallingUtil.hexToBytes(valueAsString));
    }

}

File: ios/Classes/l2capmessages/WriteL2CapChannelRequest.swift
// Copyright 2023, Continental Automotive Technologies GmbH
// All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

import Foundation

@objc
public class WriteL2CapChannelRequest: NSObject {
    
    @objc
    public let psm: Int
    @objc
    public let remoteId: String
    @objc
    public let value: Data
    
    @objc
    public init (data: NSDictionary) {
        self.psm = data[L2CapAttributeNames.keyPsm] as? Int ?? 0
        self.remoteId = data[L2CapAttributeNames.keyRemoteId] as? String ?? ""
        let valueAsString = data[L2CapAttributeNames.keyValue] as? String ?? ""
        self.value = MessageUtil.convertHexStringToBytes(hexString: valueAsString) ?? Data()
    }
                
}
